docs/blog/posts/generator.md

```diff
@@ -0,0 +1,189 @@
+Frequently Browsed Categories: [Electronics, Books, Fitness Equipment]
+Product Ratings: {Laptop: 5 stars, Wireless Headphones: 4 stars}
+Recent Search History: [best budget laptops 2023, latest sci-fi books, yoga mats]
+Preferred Brands: [Apple, AllBirds, Bench]
+Responses to Previous Recommendations: {Product123: Interested, Product456: Not Interested}
+Loyalty Program Status: Gold Member
+Average Monthly Spend: $500
+Preferred Shopping Times: Weekend Evenings
+...
+"""
+```
+We want to rank the following products for this user:
+
+```python
+products = [
+    {"product_id": 1, "product_name": "Apple MacBook Air (2023) - Latest model, high performance, portable"},
+    {"product_id": 2, "product_name": "Sony WH-1000XM4 Wireless Headphones - Noise-canceling, long battery life"},
+    {"product_id": 3, "product_name": "Apple Watch Series 7 - Advanced fitness tracking, seamless integration with Apple ecosystem"},
+    {"product_id": 4, "product_name": "Kindle Oasis - Premium e-reader with adjustable warm light"},
+    {"product_id": 5, "product_name": "AllBirds Wool Runners - Comfortable, eco-friendly sneakers"},
+    {"product_id": 6, "product_name": "Manduka PRO Yoga Mat - High-quality, durable, eco-friendly"},
+    {"product_id": 7, "product_name": "Bench Hooded Jacket - Stylish, durable, suitable for outdoor activities"},
+    {"product_id": 8, "product_name": "GoPro HERO9 Black - 5K video, waterproof, for action photography"},
+    {"product_id": 9, "product_name": "Nespresso Vertuo Next Coffee Machine - Quality coffee, easy to use, compact design"},
+    {"product_id": 10, "product_name": "Project Hail Mary by Andy Weir - Latest sci-fi book from a renowned author"}
+]
+```
+
+Let's not define our models for structured extraction. Note: instructor will conveniently let us use `Iterable` to model an iterable of our class. In this case, once we define our product recommendation model, we can slap on `Iterable` to define what we ultimately want - a (ranked) list of product recommendations. 
+
+
+```python
+import instructor
+from openai import OpenAI
+from typing import Iterable
+from pydantic import BaseModel
+
+client = instructor.patch(OpenAI(), mode=instructor.function_calls.Mode.JSON)
+
+class ProductRecommendation(BaseModel):
+    product_id: str
+    product_name: str
+    relevance_score: float
+
+Recommendations = Iterable[ProductRecommendation]
+```
+Now let's use our instructor patch. Since we don't want to wait for all the tokens to finish, will set stream to `True` and process each product recommendation as it comes in:
+```python
+
+prompt = f"Based on the following user profile:\n{profile_data}\nRank the following products from most relevant to least relevant:\n" + '\n'.join(f"{product['product_id']} {product['product_name']}" for product in products)
+
+
+recommendations_stream = client.chat.completions.create(
+    model="gpt-3.5-turbo-1106",
+    temperature=0.1,
+    response_model=Recommendations,
```

Comment:
```suggestion
    response_model=Iterable[ProductRecommendation],
```

its never reused again so i think it just makes sense to put it in the response_model